.gitignore

```diff
@@ -1,15 +1,261 @@
-# build and binary files
```

Comment:
Regarding this file, I prefer to keep only what I need in this file. I reference this same template when I need to, but my mentality is only having lines of code 100% necessary in every file. I'm happy to fix this though. If there are any specific parts of this you need for development, let me know. Curious - are you on Mac or Windows building this? Also, Xamarin or VS? I don't want to make it harder/annoying for users to submit PRs either.